test/hummingbird_test.exs

```diff
@@ -133,28 +135,65 @@ defmodule HummingbirdTest do
         expected_parent_id: expected_parent_id,
         fully_loaded_conn:
           conn(:get, "/foo")
-          |> assign(:span_id, expected_parent_id)
-          |> assign(:trace_id, expected_trace_id)
-          |> assign(:parent_id, UUID.uuid4())
+          |> put_private(:phoenix_endpoint, FooWeb.Endpoint)
+          |> put_private(:hummingbird, %{
+            span_id: expected_parent_id,
+            trace_id: expected_trace_id,
+            parent_id: UUID.uuid4(),
+            sample?: true
+          })
           |> put_req_header("x-b3-spanid", UUID.uuid4())
           |> put_req_header("x-b3-traceid", UUID.uuid4())
+          |> put_req_header("x-b3-sampled", "0")
```

Comment:
@the-mikedavis Only real concern here is that `Defer` sampling is done by absence of this header. `0` is deny and that seems ham-handed. 

It makese sense to me to have each app have ownership of how it wants to set sampling or nah, especially since it should be able to trace according to its own rules.  If it wants to make another trace, and link the two traces, that's its perogative.

Reply:
yeah this header is just for the test though. there's another line somewhere setting the default when deferred to be true or `"1"` but I agree that should be configurable on a per app basis. something like `config :hummingbird, sample_when_deferred: true|false`

Reply:
🤦 my bad

Reply:
https://github.com/NFIBrokerage/hummingbird/issues/22 made a little issue for it here